### ISSUE TITLE
Editor: Place terrain tool consumes shortcuts for tool size

### ISSUE THREAD

**Issue by frankystone** [widelands/widelands#4557]
It is not possible to use the shortcuts 1...4 to change the toolsize in the place terrain tool. Pressing one of [1,4] changes the tab in the place terrain tool.
All other tools, also those with tabs, do not have this behavior.

**To Reproduce**
1. Start editor
2. Choose place terrain tool
3. Press one of the shortcuts 1,2,3 or 4
4. The tools size do not change, instead the according tab in the place terrain tool get activated

**Expected behavior**
Tool size should have precedence over switching tabs in the place terrain tool.

**Version:**
 - OS: archlinuix
 - Widelands Version: [91caec3@master] (Debug)

**Comment by Noordfrees** [comment]
If the tool's window has focus, the tab panel will consume the keys 1-4. If the focus is on the editorinteractive (the main screen) instead, they will be used for setting the toolsize. 
A solution would be to remove the 1-9 hotkeys for tab panels, or (better) require a modifier key to activate their tab choosing function.

**Comment by frankystone** [comment]
Well, as said, this happens only for the place terrain tool, not with the other tools.

A modifier key would be good, imho. But then it should be valid for all (active) tools which have a tab panel.

**Comment by Noordfrees** [comment]
The critter tool window also shows this behaviour with the keys 1-3. The immovables tool window does *not* show it because it has 10 tabs, and the keys 1-9 are only accepted by tab panels if *all* tabs would then have a number key associated with them.

**Comment by frankystone** [comment]
The Space-key does not work under all circumstances:

1. Open Terrain tool -> Space works
2. Choose a terrain -> Space doesn't work anymore

I think the Space key should work globally, except for the Players tool.

**Comment by Noordfrees** [comment]
Whether it "works" depends on what you'd expect it to do…? Perhaps we should change the default Buildhelp hotkey (e.g. to <kbd>B</kbd>) as <kbd>Space</kbd> is also a UI selector

**Comment by frankystone** [comment]
> Whether it "works" depends on what you'd expect it to do…?
My comment was not detailed, sorry.

I would expect that pressing Space will en/disable the buildingshelp. This behavior is described in the help, as well as pressing a number sets the toolsize.

> Perhaps we should change the default Buildhelp hotkey (e.g. to B) as Space is also a UI selector

What does space do else? What do you mean with UI selector?

**Comment by Noordfrees** [comment]
<kbd>Space</kbd> is used to emulate a mouseclick on the focused UI element when using the keyboard to navigate the UI. The other UI selectors are <kbd>Enter</kbd> (click the primary button), <kbd>Tab</kbd> (next/prev element), and <kbd>Esc</kbd> (clear focus). These bindings are consistent with about every window manager I can think of. Additionally the number keys can be used to directly access the n-th entry in short lists and tab panels.

The number hotkeys for the toolsize are configurable (just like <kbd>Space</kbd> for the buildhelp). I'd be fine with changing the <kbd>1</kbd>-<kbd>9</kbd> hotkeys to something else though (e.g. how about <kbd>GUI</kbd>+(<kbd>1</kbd>-<kbd>9</kbd>) or <kbd>GUI</kbd>+(<kbd>F1</kbd>-<kbd>F12</kbd>)?)

These are the only cases where a UI selector is also used as a hotkey, so changing the default keybindings is the best approach to resolve the conflicts, IMHO.

**Comment by frankystone** [comment]
I am fine with reassigning hotkeys as long they are doing the things they are intended for.

So +1 for using a modifier + 1-9. The F-keys should only be used for _very special_ tasks, imho, e.g. reserve them for Add-ons.
